accounts/ethstore/src/account/safe_account.rs

```diff
@@ -77,16 +77,32 @@ impl SafeAccount {
 	/// Create a new `SafeAccount` from the given `json`; if it was read from a
 	/// file, the `filename` should be `Some` name. If it is as yet anonymous, then it
 	/// can be left `None`.
-	pub fn from_file(json: json::KeyFile, filename: Option<String>) -> Self {
-		SafeAccount {
+	pub fn from_file(json: json::KeyFile, filename: Option<String>) -> Result<Self, Error> {
+		SafeAccount::from_file_with_password(json, filename, &None)
+	}
+
+	/// Same as [`from_file`](#method.from_file), but with `password` which enables reading keystore files w/o address.
+	pub fn from_file_with_password(json: json::KeyFile, filename: Option<String>, password: &Option<Password>) -> Result<Self, Error> {
+		let crypto = Crypto::from(json.crypto);
+		let address = match json.address {
+			Some(address) => address.into(),
```

Comment:
imho if the `password` is provided we should validate that the `adress` is correct. Maybe switch the order here: first check if we have password and after deriving the address optionally compare it with the one in `json`.

Reply:
nice feature. It will need to throw an error when addresses do not match, but in next comment you're proposing to revert `Result`.